module/VuFind/src/VuFind/Marc/Serialization/MarcXml.php

```diff
@@ -188,4 +271,73 @@ protected static function loadXML(string $xml): \SimpleXMLElement
             throw $e;
         }
     }
+
+    /**
+     * Open a collection file
+     *
+     * @param string $file File name
+     *
+     * @return void
+     */
+    public function openCollectionFile(string $file): void
+    {
+        $this->fileName = $file;
+        $this->xml = new \XMLReader();
+        $result = $this->xml->open($file);
+        if (false === $result) {
+            throw new \Exception("Cannot open file '$file' for reading");
+        }
+        $this->currentXmlPath = [];
+    }
+
+    /**
+     * Rewind the collection file
+     *
+     * @return void;
+     */
+    public function rewind(): void
+    {
+        $this->openCollectionFile($this->fileName);
+    }
+
+    /**
+     * Get next record from the file or an empty string on EOF
+     *
+     * @return string
+     */
+    public function getNextRecord(): string
+    {
+        while ($this->xml->read()) {
```

Comment:
As above, should we validate that we have opened an XML file before assuming we can read it?

Reply:
Done.